packages/bundle-size/src/commands/uploadReport.js

```diff
@@ -0,0 +1,139 @@
+const { odata, TableClient, TablesSharedKeyCredential, TableTransaction } = require('@azure/data-tables');
+const chalk = require('chalk');
+
+const collectLocalReport = require('../utils/collectLocalReport');
+const { hrToSeconds } = require('../utils/helpers');
+
+/** @typedef {import('../utils/collectLocalReport').BundleSizeReportEntry} BundleSizeReportEntry */
+
+const AZURE_STORAGE_ACCOUNT = 'fluentbundlesize';
+const AZURE_STORAGE_TABLE_NAME = 'latest';
+const AZURE_ACCOUNT_KEY = process.env.BUNDLESIZE_ACCOUNT_KEY;
+
+/**
+ *  @param {BundleSizeReportEntry} entry
+ *
+ * @return {string}
+ */
+function createRowKey(entry) {
+  // Azure does not support slashes in "rowKey"
+  // https://docs.microsoft.com/archive/blogs/jmstall/azure-storage-naming-rules
+  return `${entry.packageName}${entry.path.replace(/.fixture.js$/, '').replace(/\//g, '')}`;
+}
+
+/**
+ * @param {typeof import('../index') & { branch: string, 'commit-sha': string }} options
+ */
+async function uploadReport(options) {
+  if (!process.env.CI) {
```

Comment:
I don't think ADO defines `process.env.CI` by default. (It only works in some of the northstar scripts because it's manually defined in various places.)
```suggestion
  if (!process.env.TF_BUILD) {
```

Reply:
As we plan we to publish this tool to run it in other repos (Tabster will be the first place to dogfood) so I decided to use [`ci-info`](https://www.npmjs.com/package/ci-info) to avoid proprietary environment variables. Done in c6ed110.